packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -1099,44 +1099,45 @@ function commitPlacement(finishedWork: Fiber): void {
   const before = getHostSibling(finishedWork);
   // We only have the top Fiber that was inserted but we need to recurse down its
   // children to find all the terminal nodes.
-  let node: Fiber = finishedWork;
-  while (true) {
-    const isHost = node.tag === HostComponent || node.tag === HostText;
-    if (isHost || (enableFundamentalAPI && node.tag === FundamentalComponent)) {
-      const stateNode = isHost ? node.stateNode : node.stateNode.instance;
-      if (before) {
-        if (isContainer) {
-          insertInContainerBefore(parent, stateNode, before);
-        } else {
-          insertBefore(parent, stateNode, before);
-        }
+  insertOrAppendPlacementNode(finishedWork, before, parent, isContainer);
+}
+
+function insertOrAppendPlacementNode(
+  node: Fiber,
+  before: ?Instance,
+  parent: Instance,
+  isContainer: boolean,
```

Comment:
This boolean never changes, so you can fork this function: `insertOrAppendPlacementNode` and `insertOrAppendPlacementNodeIntoContainer`. Then the Flow types will be correct, too.

Reply:
I thought about doing that, but was unsure about the code size increase.

Reply:
I've made the change anyway as per your suggestion :)